File: src/f90/cumulTTfrom.f90
MODULE CumulTTfrom
    USE Crop2mlModules
CONTAINS

    SUBROUTINE Calculate_CumulFROM(cumulTT, calendarMoments, calendarCumuls, switchMaize,&
            cumulTTFromZC_39, cumulTTFromZC_65, cumulTTFromZC_91)
        !      INPUT VARIABLES:
        INTEGER, INTENT(IN):: switchMaize
        REAL, INTENT(IN):: cumulTT
        REAL,DIMENSION (:), INTENT(IN):: calendarCumuls
        CHARACTER(LEN=*),DIMENSION(:), INTENT(IN):: calendarMoments
        !     OUTPUT VARIABLES:
        REAL, INTENT(OUT):: cumulTTFromZC_39, cumulTTFromZC_65, cumulTTFromZC_91
        cumulTTFromZC_39=0
        cumulTTFromZC_65=0
        cumulTTFromZC_91 =0

        IF(ANY(calendarMoments=="Anthesis")) THEN
            IF (switchMaize==0) THEN
                cumulTTFromZC_65 = cumulTT - calendarCumuls(indice(calendarMoments,"Anthesis"))
            END IF
        END IF
        IF(ANY(calendarMoments=="FlagLeafLiguleJustVisible")) THEN
            IF (switchMaize==0) THEN
                cumulTTFromZC_39 = cumulTT - calendarCumuls(indice(calendarMoments,"FlagLeafLiguleJustVisible"))
            END IF
        END IF
        IF(ANY(calendarMoments=="EndGrainFilling")) THEN
            IF (switchMaize==0) THEN
                cumulTTFromZC_91 = cumulTT-calendarCumuls(indice(calendarMoments,"EndGrainFilling"))
            END IF
        END IF
    END SUBROUTINE
END MODULE

File: src/java/CumulTTFrom.java
package pheno2;
import java.util.List;
public class CumulTTFrom
{
    public double cumulTTFromZC_65;
    public double cumulTTFromZC_39;
    public double cumulTTFromZC_91;
    public CumulTTFrom(double _cumulTTFromZC_65,double _cumulTTFromZC_39,double _cumulTTFromZC_91)
    {
        this.cumulTTFromZC_65=_cumulTTFromZC_65;
        this.cumulTTFromZC_39=_cumulTTFromZC_39;
        this.cumulTTFromZC_91=_cumulTTFromZC_91;
    }
}

class Estimation_CumulTTFrom
{
    public static CumulTTFrom CalculateCumulTTFrom(List<String> calendarMoments,List<Double> calendarCumuls,int switchMaize,double cumulTT)
    {


/*
     CumulTTFrom Model

    Author: 
    Reference: Modeling development phase in the 
                Wheat Simulation Model SiriusQuality.
                See documentation at http://www1.clermont.inra.fr/siriusquality/?page_id=427
    Instituton: INRA Montpellier
    Abstract: Calculate CumulTT 
    
*/
        double cumulTTFromZC_65;
        double cumulTTFromZC_39;
        double cumulTTFromZC_91;

        cumulTTFromZC_65 = 0.0D;
        cumulTTFromZC_39 = 0.0D;
        cumulTTFromZC_91 = 0.0D;
        if (calendarMoments.contains("Anthesis")){
            if (switchMaize == 0)
                cumulTTFromZC_65 = cumulTT-calendarCumuls.get(calendarMoments.indexOf("Anthesis"));
        }    
        if (calendarMoments.contains("FlagLeafLiguleJustVisible")){
            if (switchMaize == 0)
                cumulTTFromZC_39 = cumulTT-calendarCumuls.get(calendarMoments.indexOf("FlagLeafLiguleJustVisible"));
        }   
        if (calendarMoments.contains("EndGrainFilling")){
            if (switchMaize == 0)
                cumulTTFromZC_91 = cumulTT-calendarCumuls.get(calendarMoments.indexOf("FlagLeafLiguleJustVisible"));
        }
        return new CumulTTFrom(cumulTTFromZC_65,cumulTTFromZC_39,cumulTTFromZC_91);
    }

}
